From 588b55402897c247232321df1353642f048d1d11 Mon Sep 17 00:00:00 2001
From: Monika Korada <quic_koramoni@quicinc.com>
Date: Mon, 10 Jun 2024 16:46:48 +0530
Subject: [PATCH] KW: wifi: mac80211: fix NULL pointer access Klocwork issue

Currently, we are dereferencing params->chan pointer without
a NULL check in ieee80211_mgmt_tx function.
Fix this issue by adding a NULL check.

Patch-dependency: none
Patch-work: none
Fixes: a2fcfccbad43 (mac80211: move off-channel/mgmt-tx code to offchannel.c)
Note: ath.git commit id is mentioned in the Fixes tag

Signed-off-by: Monika Korada <quic_koramoni@quicinc.com>
---
 net/mac80211/offchannel.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

--- a/net/mac80211/offchannel.c
+++ b/net/mac80211/offchannel.c
@@ -1033,11 +1033,16 @@ int ieee80211_mgmt_tx(struct wiphy *wiph
 			local->hw.offchannel_tx_hw_queue;
 
 	/* This will handle all kinds of coalescing and immediate TX */
+	if (!params->chan) {
+		ret = -EINVAL;
+		goto out_unlock;
+	}
 	ret = ieee80211_start_roc_work(local, sdata, params->chan,
 				       params->wait, cookie, skb,
 				       IEEE80211_ROC_TYPE_MGMT_TX);
 	if (ret)
 		ieee80211_free_txskb(&local->hw, skb);
+
  out_unlock:
 	mutex_unlock(&local->mtx);
 	return ret;
